core/shared/src/main/scala/zio/Task.scala

```diff
@@ -251,6 +251,12 @@ object Task {
   final def foldLeft[S, A](in: Iterable[A])(zero: S)(f: (S, A) => Task[S]): Task[S] =
     ZIO.foldLeft(in)(zero)(f)
 
+  /**
+   * @see See [[zio.ZIO.foldRight]]
+   */
+  final def foldRight[S, A](in: Iterable[A])(zero: S)(f: (A, S) => Task[S]): Task[S] =
```

Comment:
Need to add versions of other new methods here and in other companion objects where appropriate (i.e. only for data types that can fail).

Reply:
Don't we still need `validateM` and `validateFirstM` here?

Reply:
`validateM` and `validateFirstM`error type is not Throwable. It is List[Throwable]. I do not think it will work out

Reply:
You're right. The return type would have to be `IO[List[Throwable], List[B]]`. It could be useful but it does break the contract that methods on the companion object return the same type. Fine to leave out.